File: backend/app/models/issue_type.py
from sqlalchemy import String, ForeignKey, Boolean, Integer, BigInteger
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base, AuditMixin, SoftDeleteMixin

class IssueType(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "issue_types"
    
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    color: Mapped[str] = mapped_column(String(50), default="#6B7280")
    icon: Mapped[str] = mapped_column(String(50), default="circle")
    is_default: Mapped[bool] = mapped_column(Boolean, default=False)
    sequence: Mapped[int] = mapped_column(Integer, default=1)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    
    project_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("projects.id"), nullable=False)
    workspace_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("workspaces.id"), nullable=False)
    
    project: Mapped["Project"] = relationship(back_populates="issue_types")
    workspace: Mapped["Workspace"] = relationship()

File: backend-go/internal/model/issue_type.go
package model

// IssueType represents a work item type (e.g., Bug, Task, Story, Epic).
// Workspace-scoped with optional project_id — when nil, the type is shared across all workspace projects.
type IssueType struct {
	BaseModel
	Name        string  `gorm:"type:varchar(100);not null" json:"name"`
	Color       string  `gorm:"type:varchar(20);default:'#6366F1'" json:"color"`
	Icon        string  `gorm:"type:varchar(50);default:'circle'" json:"icon"`
	IsDefault   bool    `gorm:"default:false" json:"is_default"`
	Sequence    int     `gorm:"default:1" json:"sequence"`
	IsActive    bool    `gorm:"default:true" json:"is_active"`
	ProjectID   *uint64 `gorm:"index" json:"project_id"`
	WorkspaceID uint64  `gorm:"not null;index" json:"workspace_id"`

	// Relationships
	Workspace    Workspace        `gorm:"foreignKey:WorkspaceID" json:"-"`
	FieldLinks   []IssueTypeField `gorm:"foreignKey:TypeID" json:"-"`
}

func (IssueType) TableName() string {
	return "issue_types"
}

// IssueTypeField is a join table for many-to-many association between IssueType and CustomField.
type IssueTypeField struct {
	TypeID     uint64 `gorm:"primaryKey;autoIncrement:false" json:"type_id"`
	FieldID    uint64 `gorm:"primaryKey;autoIncrement:false" json:"field_id"`
	IsRequired bool   `gorm:"default:false" json:"is_required"`
	Sequence   int    `gorm:"default:1" json:"sequence"`

	IssueType  IssueType   `gorm:"foreignKey:TypeID;constraint:OnDelete:CASCADE" json:"-"`
	Field      CustomField `gorm:"foreignKey:FieldID;constraint:OnDelete:CASCADE" json:"-"`
}

func (IssueTypeField) TableName() string {
	return "issue_type_fields"
}
